Message Corrections & Refactoring

MacSplody opened the pull request:
#460 
#462

MacSplody committed:
Correct Aliases used for Patch Messages

Alias hasPluginPatch will be used when patches are provided on the mods
page in a seperate download.
The alias haspatchIncluded will be used for patches included with the
plugins installer.

Seperate Obsidian 1st & 3rd party patch links:
These should be kept separate to provide the user with both options,
as they have different approaches to patching these mods.
Both will still only show if the other is not present.

Change True Storm Merged Compatibility patch message to 3rd party:
The Merged Compatibility patches are provided on a seperate page by a
third party and are not supported by the main authors.

MacSplody committed:
Refactor Cleaning Messages

Remove Clean entries with deleted nav:
When a DLC is still dirty (e.g it has deleted nav) it will be left blank, the same is done for Fallout 4, and I think it's important to be consistent, especially with cleaning.

Reworded QAC Dirty messages:
Requires manual fix now includes a link with more information on deleted navmeshes.
Shortened Quickclean message, any information on QAC can be provided on the linked guide page.

Seperate Wild edit message and cleaning guide link:
I've seperated these as sometimes a mod has wild edits, but no cleaning guide is available.
This way LOOT can still inform the user of possible issues, even if a fix is not provided.
I think it's a better approach to always provide a warning if wild edits are present.
And to provide a linked cleaning guide if available.
Remove Built in cleaning guides:
These in built cleaning messages feel very clunky and seem to provide more confusion than help.
The webpage guides we have links to provide better formatting, which are much easier to follow.

MacSplody committed:
Refactor Patch Provided Message

Replaced hasPluginPatch with patchProvided.

To be used in situations where a patch is provided on the plugins mod page.
Removed sub link to plugins own mod page as it seems redundant when a link is already provided in locations.
The name of and optionally a link to the Conflicting mod will still be provided.

MacSplody committed:
Refactor Patch Included Message

Changed Alias and patch message to be more consistent with the language used in patch provided message.
Removed redundant sub with modpage link, as pointing to the plugins own modpage seems unnecessary when the patch is provided with the installer.

MacSplody committed:
Add Common entry for 3rd party patches

Added common entry message for patches provided by a 3rd party.
Patches are provided on a seperate page by a third party and are not supported by the main authors.
I think it's important to make that distinction clear to avoid any issues with support.

MacSplody committed:
Refactor Compatibility Note Messages

Replace Compatibility Notes with common entry.
Remove messages covered by Compatibility Notes.

MacSplody committed:
Fix VW & SOS Duplicate Patch Warning

Vivid Weathers & Sounds of Skyrim both provide their own patch.

Check that both patches are inactive before displaying patch message.
Add useOnlyOneX instead of incompatible message.

https://github.com/loot/skyrimse/issues/462
https://github.com/loot/loot/issues/1050

Ortham commented on masterlist.yaml:
```diff
@@ -28,24 +28,26 @@ common:
-      - lang: de
-        text: "Sie nutzen anscheinend %1%, haben aber nicht den %1% Patch für dieses Plugin aktiviert: %2%"
-  - &hasPatchIncluded
+        text: 'You seem to be using %1%, but you have not enabled a compatibility patch for this mod. A compatibility patch is provided on this plugins mod page.'
```
```suggestion
        text: 'You seem to be using %1%, but you have not enabled a compatibility patch for this mod. A compatibility patch is provided on this plugin's mod page.'
```

Ortham commented on masterlist.yaml:
```diff
@@ -28,24 +28,26 @@ common:
-        text: "A patch is included with the %2% installer, please make sure you have installed the %1% patch and have it activated."
-      - lang: de
-        text: "Ein Patch ist in dem %2% Installer enthalten, bitte stellen Sie sicher, dass Sie den %1% Patch installiert und aktiviert haben."
+        text: 'You seem to be using %1%, but you have not enabled a compatibility patch for this mod. A compatibility patch is included with this plugins installer.'
```
```suggestion
        text: 'You seem to be using %1%, but you have not enabled a compatibility patch for this mod. A compatibility patch is included with this plugin's installer.'
```

Ortham commented on masterlist.yaml:
```diff
@@ -28,24 +28,26 @@ common:
+    type: say
+    content:
+      - lang: en
+        text: 'It is recommended that you read this mods [Compatibility Notes](%1%).'
```
```suggestion
        text: 'It is recommended that you read this mod's [Compatibility Notes](%1%).'
```

Ortham commented on masterlist.yaml:
```diff
@@ -139,22 +141,26 @@ common:
       - lang: de
-        text: "Dieses Plugin enthält unbeabsichtigte Änderungen. Eine manuelle [SSEEdit](https://www.nexusmods.com/skyrimspecialedition/mods/164)-Säuberungsanleitung ist hier verfügbar: %1%"
+        text: 'Dieses Plugin enthält wilde Modifikationen weit über ITM-, und UDR Einträge und entfernte Navmeshes hinaus und benötigt vermutlich mehr manuelle Säuberung um andere Mods nicht zu beeinträchtigen.'
+  - &cleaningGuide
```
Can this message not be appended to the `info` text for the cleaning metadata? It would keep all the relevant info in one place any maybe simplify the masterlist. Or combined with the `wildEdits` message, I don't know how many plugins have one without the other.

MacSplody replied:
Same as below
>I could make a post in AFK forums for wild edit cleaning guides, which would be separate to the >LOOT page. I'm not sure where else would be appropriate to host them.

MacSplody replied:
@Ortham I'm going to go with appending the guides to the wildedits message as a sub.
![image](https://user-images.githubusercontent.com/1944639/48585165-aff50e00-e923-11e8-9dfb-efe9317d732e.png)

I've tried adding it to the cleaning info messages, but you'll still have the `xEdit found dirty edits` message, which seems out of place since it's manual cleaning.

MacSplody replied:
Updated in https://github.com/loot/skyrimse/pull/468/commits/bbf821e16285ff6a1764b402bea99665fedb6fa4

Ortham commented on masterlist.yaml:
```diff
@@ -4956,16 +4693,7 @@ plugins:
-        subs:
-          - |+
-
-              #### Remove the following records with xEdit
```
If this information isn't going to be included in a message, it should probbaly be made available somewhere else and linked to. Unless it's inaccurate?

MacSplody replied:
I haven't had a chance to check if it's still relevant. I could make a post in AFK forums for wild edit cleaning guides, which would be separate to the LOOT page. I'm not sure where else would be appropriate to host them.

Ortham replied:
Some plugins have links to a Creation Kit wiki page, IIRC, you could put the info there.

MacSplody replied:
Creation Kit wiki page is good, I'll set up an account and add the guides there.

MacSplody replied:
Okay tried the wiki and the captcha makes it a pain.
I could use a github page with markdown.
https://cptmcsplody.github.io/Guides/

CptMcSplody replied:
I've added the guides here https://cptmcsplody.github.io/Guides/ for the moment, at least before I translate them to the Wiki format.

CptMcSplody replied:
I've added the guides here https://cptmcsplody.github.io/X-Guides/

MacSplody replied:
I've added the guides here https://cptmcsplody.github.io/xGuides/

MacSplody replied:
Resolved in https://github.com/loot/skyrimse/commit/2acab29df606cad6755c3d5eb565de684ee322d5

Ortham commented on masterlist.yaml:
```diff
@@ -4981,16 +4709,7 @@ plugins:
-        subs:
-          - |+
-
-              #### Remove the following records with xEdit
```
Same as above, removing this loses information.

Ortham commented on masterlist.yaml:
```diff
@@ -5006,19 +4725,7 @@ plugins:
-        subs:
-          - |+
-
-              #### Edit the following records with xEdit
```
Same as above, removing this loses information.

Ortham commented on masterlist.yaml:
```diff
@@ -5034,19 +4741,7 @@ plugins:
     inc:
       - 'TheWulfPanda - Faction Teleport Spells.esp'
     msg:
-      - <<: *hasWildEdits
```
Same as above, removing this loses information.

Ortham commented on masterlist.yaml:
```diff
@@ -5062,15 +4757,7 @@ plugins:
-        subs:
-          - |+
-
-            #### Remove the following records with xEdit
```
Same as above, removing this loses information.

Ortham commented on masterlist.yaml:
```diff
@@ -5086,15 +4773,7 @@ plugins:
     inc:
       - 'TheWulfPanda - Teleport Spells Towns and Villages.esp'
     msg:
-      - <<: *hasWildEdits
```
Same as above, removing this loses information.

Ortham commented on masterlist.yaml:
```diff
@@ -5328,17 +5001,7 @@ plugins:
-        subs:
-          - |+
-
-              #### Description of Wild Edit
```
Same as above, removing this loses information.

Ortham commented on masterlist.yaml:
```diff
@@ -5218,20 +4897,14 @@ plugins:
-        subs:
-          - 'Convenient Horses'
-          - '[Bruma patch for Convenient Horses](https://www.nexusmods.com/skyrimspecialedition/mods/13812/)'
+      - <<: *patchProvided
```
The patch isn't provided at <https://www.nexusmods.com/skyrimspecialedition/mods/10917/>...

Ortham commented on masterlist.yaml:
```diff
@@ -5218,20 +4897,14 @@ plugins:
-        subs:
-          - 'Flora Respawn Fix'
-          - '[Flora Respawn Fix SE for Beyond Skyrim](https://www.nexusmods.com/skyrimspecialedition/mods/11072/)'
+      - <<: *patchProvided
```
The patch isn't provided at <https://www.nexusmods.com/skyrimspecialedition/mods/10917/>...

MacSplody replied:
I forgot about these thanks.

MacSplody replied:
Fixed in https://github.com/loot/skyrimse/pull/468/commits/bbf821e16285ff6a1764b402bea99665fedb6fa4

Ortham commented on masterlist.yaml:
```diff
@@ -5218,20 +4897,14 @@ plugins:
-        subs:
-          - 'Ordinator - Perks of Skyrim'
-          - '[Ordinator Beyond Skyrim Bruma Patch](https://www.nexusmods.com/skyrimspecialedition/mods/10934/)'
+      - <<: *patchProvided
```
The patch isn't provided at <https://www.nexusmods.com/skyrimspecialedition/mods/10917/>...

MacSplody replied:
Fixed in https://github.com/loot/skyrimse/pull/468/commits/bbf821e16285ff6a1764b402bea99665fedb6fa4

Ortham commented on masterlist.yaml:
```diff
@@ -5694,15 +5357,7 @@ plugins:
-        subs:
-          - |+
-
-              - Open SSEEdit and load **Distinct Interiors.esp**.
```
Same as above, removing this loses information.

MacSplody replied:
https://github.com/loot/skyrimse/commit/2acab29df606cad6755c3d5eb565de684ee322d5

Ortham commented on masterlist.yaml:
```diff
@@ -5741,14 +5396,7 @@ plugins:
-        subs:
-          - |+
-
-              - Open SSEEdit and load **Distinct Interiors - Player Homes.esp**.
```
Same as above, removing this loses information.

MacSplody replied:
https://github.com/loot/skyrimse/commit/2acab29df606cad6755c3d5eb565de684ee322d5

Ortham commented on masterlist.yaml:
```diff
@@ -6158,75 +5736,47 @@ plugins:
-        subs:
-          - 'Dawn of Skyrim'
-          - '[JKs Skyrim_Dawn of Skyrim_Patch.esp](https://www.nexusmods.com/skyrimspecialedition/mods/16852/)'
+      - <<: *patchProvided
```
The patch isn't at https://www.nexusmods.com/skyrimspecialedition/mods/6289/

MacSplody replied:
Fixed in https://github.com/loot/skyrimse/pull/468/commits/bbf821e16285ff6a1764b402bea99665fedb6fa4

Ortham reviewed (changes requested):
Generally good, though a few mistakes sneaked in and there's a bit of information loss about wild edits.

MacSplody committed:
Pull Request Corrections

Language correction for common patch messages.
Patch Provided Links updated to 3rd party messages.
Appended Cleaning Guide Links to Wild Edits Message as optional Sub.

Co-Authored-By: Oliver Hamlet <oliver.hamlet@gmail.com>

MacSplody committed:
Pull Request Corrections 2

Added links for external cleaning guides.

Seperated Wild edit cleaning & Deleted navmesh cleaning.

If available, guides for cleaning Deleted navmesh will be passed using the info key of reqManualFix.
